mephisto/data_model/task.py

```diff
@@ -363,9 +363,12 @@ def get_assignment_statuses(self) -> Dict[str, int]:
             for status in AssignmentState.valid()
         }
 
+    def update_assignment_generator_done(self, status) -> None:
+        self.assignments_generator_done = False
+
     def update_completion_progress(self, task_launcher) -> None:
         """ Flag the task run that the assignments' generator has finished """
-        if task_launcher.get_assignment_thread_flag():
+        if task_launcher.get_assignments_are_all_created():
             self.assignments_generator_done = True
```

Comment:
Why are these separate functions? You can define a python function that takes in keyword arguments (like `status` and `task_launcher` but defaults these to `None`) and chooses behavior based on which arguments are provided. Alternately, you can make `update_completion_progress` rely on `update_assignment_generator_done`.

Further, `update_assignment_generator_done`'s function signature at the moment doesn't actually represent what you're doing with it.

Reply:
fixed it now. Thanks!